api/worker_grpc.go

```diff
@@ -563,73 +563,83 @@ func NotifyWorkers(daoWrapper dao.DaoWrapper) func() {
 				sentry.CaptureException(err)
 				continue
 			}
-			sources := []string{lectureHallForStream.CombIP, lectureHallForStream.PresIP, lectureHallForStream.CamIP}
-			for sourceNum, source := range sources {
-				if source == "" {
-					continue
-				}
-				var sourceType string
-				if sourceNum == 1 {
-					sourceType = "PRES"
-				} else if sourceNum == 2 {
-					sourceType = "CAM"
-				} else {
-					sourceType = "COMB"
-				}
-				server, err := daoWrapper.IngestServerDao.GetBestIngestServer()
-				if err != nil {
-					log.WithError(err).Error("Can't find ingest server")
-					continue
-				}
-				var slot model.StreamName
-				if sourceType == "COMB" { //try to find a transcoding slot for comb view:
-					slot, err = daoWrapper.IngestServerDao.GetTranscodedStreamSlot(server.ID)
-				}
-				if sourceType != "COMB" || err != nil {
-					slot, err = daoWrapper.IngestServerDao.GetStreamSlot(server.ID)
+
+			createStreamRequest := func(sourceType string, source string) {
```

Comment:
This does not need to be an inline function. Can you extract it to package level?

Reply:
Done.